### PR TITLE
Offer to pay after missed deadline

### LINKED ISSUE #3680: Dashboard: When learner misses payment deadline, tell them when they can pay next
A learner who has passed an FA course, but missed the payment deadline, can pay for the course the next time an exam is offered. Instead of telling them to enroll again, we should let them know when they can next pay. 

This may require adding another field to the course or exam model to indicate when we will start taking payments again. 

See https://odl.zendesk.com/agent/tickets/10417 for an example.

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #3680

#### What's this PR do?
Add new states for `missed-deadline` status if exams are present.

#### How should this be manually tested?
Run 
`scripts/test/run_snapshot_dashboard_states.sh --match "create_passed_missed_payment_for_exam"`

### CODE REVIEW

File: dashboard/api.py [hunk 1]
```diff
@@ -510,6 +511,23 @@ def get_future_exam_runs(course):
             order_by('date_first_schedulable').values_list('date_first_schedulable', flat=True))
 
 
+def get_past_recent_exam_run(course):
+    """
+    Return a recent exam run
```
Comment (amir-qayyum-khan) [inline dashboard/api.py]: it is returning a string

File: dashboard/api.py [hunk 2]
```diff
@@ -510,6 +511,23 @@ def get_future_exam_runs(course):
...
+
+    Args:
+        course (courses.models.Course): A course
+
+    Returns:
+        tuple: a list of dates when future exams become schedulable
```
Comment (amir-qayyum-khan) [inline dashboard/api.py]: it is returning a date range i.e start and end date of a last schedulable exam run? Am i right. Can we make it more clear?

**Comment by annagav** [comment]
@amir-qayyum-khan, There was a merge conflict so I squashed all commits to make it easier. I addressed your comments.

**Comment by annagav** [comment]
@amir-qayyum-khan, I fixed the tests.

**Comment by amir-qayyum-khan** [comment]
is there code change? why test were failing?

**Comment by annagav** [comment]
The tests were failing from the merge conflict.

**Pull request merged**